src/injectStyled.js

```diff
@@ -0,0 +1,26 @@
+import {createElement} from 'react'
+
+import composeClasses from './utils/compose-classes'
+import type {styledType} from './types'
+
+const injectStyled = (styled: styledType) => (InnerComponent: ReactClass<any>) => {
+  styled.mountSheets()
+
+  const {sheets} = styled
+  const {staticSheet, dynamicSheet} = sheets
+
+  const classNames = new Set([
+    ...Object.keys(staticSheet.classes),
+    ...Object.keys(dynamicSheet.classes)
+  ])
+
+  const classes = [...classNames]
+    .reduce((acc, name) => ({
+      ...acc,
+      [name]: composeClasses(staticSheet.classes[name], dynamicSheet.classes[name]),
+    }), {})
+
+  return (...props: any) => createElement(InnerComponent, {sheets, classes, ...props})
```

Comment:
I would probably not inject `sheets` at all. What use cases are there anyways? Also in react-jss I inject `sheet` which refs to dynamicSheet if there is one and to static if there is no. Actually I did it because it was already there for the case people still using props.sheet. From my experience there should be no need to use sheet object in react.